File: strutsAnnotation/src/main/java/org/softlang/basics/Employee.java
package org.softlang.basics;

import java.io.Serializable;

/**
 * An employee has a salary and some person information
 * 
 */
public class Employee extends Subunit implements Serializable {

	private static final long serialVersionUID = -210889592677165250L;

	private Long id;
	private String name;
	private String address;
	private double salary;

	
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
		
	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}
	

	public double getSalary() {
		return salary;
	}

	public void setSalary(double salary) {
		this.salary = salary;
	}

	public double getTotal() {
		return getSalary();
	}	
	
	public void cut() {
		setSalary(getSalary() / 2);
	}

	@Override
	public boolean isDepartment() {
		return false;
	}	
}

File: qt/QtImplementation/employee.cpp
#include "employee.h"

Employee::Employee(){
    Employee("unknown","unknown",0.0);
}

Employee::Employee(QString name, QString address, double salary)
{
    this->address = address;
    this->name = name;
    this->salary = salary;
}

QString Employee::getName()
{
    return name;
}

void Employee::setName(QString name)
{
    this->name = name;
}

QString Employee::getAddress()
{
    return address;
}

void Employee::setAddress(QString address)
{
    this->address = address;
}

double Employee::getSalary()
{
    return salary;
}

void Employee::setSalary(double salary)
{
    this->salary = salary;
}

void Employee::cut()
{
    this->salary = salary/2.0;
}
